src/mixins/cloneSelection.js

```diff
@@ -42,12 +42,18 @@ export default {
             clonedDataOutputAssociations.push(clonedFlow);
             clonedNodes.push(clonedFlow);
           } else {
+            // Validate boundary events and collect in clonedBoundaryEvents array
             const clonedElement = this.cloneElementAndCalculateOffset(node);
-            clonedNodes.push(clonedElement);
+            if (node.definition && node.definition.$type !== 'bpmn:BoundaryEvent') {
+              clonedNodes.push(clonedElement);
+            }  else {
+              clonedBoundaryEvents.push(clonedElement);
+            }
           }
         });
       }
-
+      // Sets the clonedBoundaryEvents at the end of the array
+      clonedNodes = [...clonedNodes, ...clonedBoundaryEvents];
```

Comment:
Does the order affect how the cloned elements are connected afterwards? I think that maybe it is not necessary to put it at the end of the array and would be better to check the code that connects flows before adding every clone to the canvas.

Thoughts? @josechirivella @caleeli?

Reply:
@rodriquelca commented me the boundary clone needs the task clone to exists before to attach it.